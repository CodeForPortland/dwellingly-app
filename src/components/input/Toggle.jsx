import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Toggle extends Component {
  constructor(props) {
    super(props);

    this.handleToggle = this.handleToggle.bind(this);

    this.state = {
      toggle: this.props.value
    };
  }

  handleToggle() {
    this.setState(
      ({ toggle }) => ({
        toggle: !toggle
      }),
      () => {
        this.props.onToggle(this.props.model, this.state.toggle);
      }
    );
  }

  render() {
    const { blockClass, id, label } = this.props;
    return (
      <div className="inline-input">
        <label htmlFor={id}>
          <span className={`${blockClass}__label`}>{label}</span>
          <div className="inline-input__toggle">
            <input type="checkbox" checked={this.state.toggle} id={id} />
            <button
              aria-label="Checkbox toggle"
              className="inline-input__toggle__btn"
              onClick={this.handleToggle}
              type="button"
            />
          </div>
        </label>
      </div>
    );
  }
}

Toggle.propTypes = {
  blockClass: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  model: PropTypes.string,
  onToggle: PropTypes.func,
  value: PropTypes.bool
};

Toggle.defaultProps = {
  model: undefined,
  onToggle: () => {},
  value: false
};

export default Toggle;
