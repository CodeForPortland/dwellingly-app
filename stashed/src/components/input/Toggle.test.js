import React from 'react';
import renderer from 'react-test-renderer';
import Toggle from './Toggle';

describe('Toggle Component', () => {
  it('correctly renders Input component', () => {
    const ToggleComponent = renderer.create(<Toggle />).toJSON();
    expect(ToggleComponent).toMatchSnapshot();
  });
  it('toggles state on button click', () => {
    const wrapper = shallow(<Toggle />)
    const toggleButton = wrapper.find('button');
    toggleButton.simulate('click');
    expect(wrapper.state().toggle).toEqual(true);
  });
  it('toggles state on checkbox select', () => {
    const wrapper = shallow(<Toggle />)
    const toggleCheckbox = wrapper.find('input');
    toggleCheckbox.simulate('change');
    expect(wrapper.state().toggle).toEqual(true);
  });
});
