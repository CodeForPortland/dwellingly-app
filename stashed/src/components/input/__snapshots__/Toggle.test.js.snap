// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Component correctly renders Input component 1`] = `
<label
  className="inline-input__toggle"
>
  <span
    className="undefined__label"
  />
  <div
    className="switch"
  >
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    <button
      aria-label="Checkbox toggle"
      className="switch__btn"
      onClick={[Function]}
      type="button"
    />
  </div>
</label>
`;
